### PR TITLE
Enable --auto-loss-scaling flag in Optimum

### PR DESCRIPTION
**Pull request by sergiopperez**
# What does this PR do?

Enables #128 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests?

### CODE REVIEW

File: optimum/graphcore/training_args.py [hunk 1]
```diff
@@ -432,6 +436,11 @@ def __post_init__(self):
         else:
             self.gradient_accumulation_steps = 1
 
+        if self.auto_loss_scaling is True:
+            override_str.append(f"auto_loss_scaling={self.auto_loss_scaling}")
+        else:
+            self.auto_loss_scaling = False
```
Comment (jimypbr) [inline optimum/graphcore/training_args.py:442]: this is unnecessary since it's default false

File: optimum/graphcore/training_args.py [hunk 2]
```diff
@@ -432,6 +436,11 @@ def __post_init__(self):
         else:
             self.gradient_accumulation_steps = 1
 
+        if self.auto_loss_scaling is True:
```
Comment (jimypbr) [inline optimum/graphcore/training_args.py:439]: if self.auto_loss_scaling is enough

Comment (sergiopperez) [reply optimum/graphcore/training_args.py:439]: Thanks James, so to clarify, are you suggesting to
1) Remove
```
 if self.auto_loss_scaling is True:
      override_str.append(f"auto_loss_scaling={self.auto_loss_scaling}")
else:
      self.auto_loss_scaling = False
```

or 2) Modify it as 
```
if self.auto_loss_scaling:
      override_str.append(f"auto_loss_scaling={self.auto_loss_scaling}")
else:
      self.auto_loss_scaling = False
```

Comment (jimypbr) [reply optimum/graphcore/training_args.py]: What you did in the commit is right

**Comment by sergiopperez** [comment]
@jimypbr @gejinchen this PR is ready to be reviewed — let me know your thoughts about it. Apart from the commits, I've checked that ALS converges fine for the pretraining of BERT large and GPT2 small.

**Comment by jimypbr** [comment]
Thanks great stuff

**Pull request merged**